src/api/define.js

```diff
@@ -102,10 +102,19 @@ function generateUniqueName(name) {
 }
 
 function registerUniqueName(name) {
-  registry[name] = registry[name] ? registry[name] + 1 : 1;
+  if (registry[name]) {
+    registerUniqueName(name + '-'+ registry[name]);
```

Comment:
The problem is that it doesn't do what it suppose to do, that's all. It just bugs me. For example, those tests would fail with the previous implementation: https://github.com/skatejs/skatejs/pull/674/commits/b1072eb7a9b134383eb279b0f6defd89645ec44b#diff-e52899dbc88c4ba1c1db13686955c644R27.

And it's quite easy to imagine a practical situation based on those tests. For example:
<test-list>
  <test-item></test-item>
  <test-item-2></test-item-2>  
</test-list>
This would fail with HMR. And to make it worth fixing - it would fail unexpectedly for those who doesn't know how exactly skate works from the inside, i.e. for the outside developers. 

BTW, speaking of HMR and recursion. With every reload there would be registered x new components, all of which cannot be unregistered, isn't it? If this is okay, can we just instead of recursion register a new component every time with some unique hash instead of numbers? Using something like this: https://github.com/broofa/node-uuid.